core/src/main/java/org/elasticsearch/common/lucene/index/ESIndexDeletionPolicy.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.lucene.index;
+
+import org.apache.lucene.index.IndexCommit;
+import org.apache.lucene.index.IndexDeletionPolicy;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * An {@link IndexDeletionPolicy} that deletes unneeded index commits, and returns index commits are not deleted by this policy.
+ */
+public interface ESIndexDeletionPolicy {
```

Comment:
do we need a separate interface? If we want to know which commits can be kept,
we can just do `commits.stream().allMatch(c -> c.isDeleted() == false)` after invoking the `onInit` or `onCommit` methods?

Reply:
This interface is to avoid keeping translog of snapshotted index commits. `SnapshotDeletionPolicy` suppresses the `delete()` for snapshotted commits.

https://github.com/apache/lucene-solr/blob/e2521b2a8baabdaf43b92192588f51e042d21e97/lucene/core/src/java/org/apache/lucene/index/SnapshotDeletionPolicy.java#L221-L225

Reply:
I should have documented its purpose.